web/src/navigator.rs

```diff
@@ -35,68 +35,44 @@ impl WebNavigatorBackend {
             upgrade_to_https,
         }
     }
-
-    fn run_script(&self, js_code: &str) {
-        if self.allow_script_access {
-            let window = web_sys::window().expect("window()");
-            let document = window.document().expect("document()");
-            let body = document.body().expect("body()");
-
-            let script = document.create_element("script").unwrap();
-            script.set_inner_html(&js_code);
-
-            let _ = body.append_child(&script);
-            let _ = body.remove_child(&script);
-        } else {
-            log::error!("SWF tried to run a script, but script access is not allowed");
-        }
-    }
 }
 
 impl NavigatorBackend for WebNavigatorBackend {
     fn navigate_to_url(
         &self,
-        url: String,
+        raw_url: String,
         window_spec: Option<String>,
         vars_method: Option<(NavigationMethod, IndexMap<String, String>)>,
     ) {
-        const JAVASCRIPT_PREFIX: &str = "javascript:";
+        // If the URL is empty, we ignore the request
+        if raw_url.is_empty() {
+            return;
+        }
 
         if let Some(window) = window() {
-            let url_trimmed = url.trim();
-            if url_trimmed
-                .get(..JAVASCRIPT_PREFIX.len())
-                .unwrap_or_default()
-                .eq_ignore_ascii_case(JAVASCRIPT_PREFIX)
-            {
-                let target = window_spec.unwrap_or_else(|| "".to_string());
-                if target.is_empty() || target == "_self" || target == "undefined" {
-                    self.run_script(&url);
+            let url;
+            let document = window.document().expect("Could not get document");
+            if let Ok(Some(base_uri)) = document.base_uri() {
```

Comment:
On a second thought, even `resolve_relative_url` can be used.

Reply:
Yeah I agree, that would be much cleaner

Reply:
Actually, resolve_relative_url returns a string, not the parsed URL, so I would have to reparse it, which I don't think is something we should do